sdk/src/main/java/io/opentelemetry/sdk/trace/RecordEventsReadableSpan.java

```diff
@@ -318,8 +317,34 @@ public void setAttribute(String key, boolean value) {
   public void setAttribute(String key, AttributeValue value) {
     Preconditions.checkNotNull(key, "key");
     Preconditions.checkNotNull(value, "value");
-    if (value.getType() == Type.STRING && StringUtils.isNullOrEmpty(value.getStringValue())) {
-      return;
+    switch (value.getType()) {
+      case STRING:
+        if (StringUtils.isNullOrEmpty(value.getStringValue())) {
+          return;
+        }
+        break;
+      case STRING_ARRAY:
+        if (value.getStringArrayValue().size() == 0) {
+          return;
+        }
+        break;
+      case BOOLEAN_ARRAY:
+        if (value.getBooleanArrayValue().size() == 0) {
+          return;
+        }
+        break;
+      case LONG_ARRAY:
+        if (value.getLongArrayValue().size() == 0) {
+          return;
+        }
+        break;
+      case DOUBLE_ARRAY:
+        if (value.getDoubleArrayValue().size() == 0) {
+          return;
+        }
+        break;
+      default:
```

Comment:
Is this required?

Reply:
Yes, otherwise the checkstyle plugin will fail the build